geomesa-datastore-bundle/geomesa-datastore-processors/src/main/scala/org/geomesa/nifi/datastore/processor/AvroIngestProcessor.scala

```diff
@@ -11,30 +11,94 @@ package org.geomesa.nifi.datastore.processor
 
 import java.io.InputStream
 
+import org.apache.nifi.components.PropertyDescriptor
 import org.apache.nifi.flowfile.FlowFile
 import org.apache.nifi.processor._
 import org.apache.nifi.processor.io.InputStreamCallback
+import org.geomesa.nifi.datastore.processor.AbstractGeoIngestProcessor.checkCompatibleSchema
+import org.geomesa.nifi.datastore.processor.AvroIngestProcessor.{AvroMatchMode, buildWriter}
 import org.geotools.data._
+import org.geotools.feature.simple.SimpleFeatureBuilder
+import org.geotools.util.Converters
 import org.locationtech.geomesa.features.avro.AvroDataFileReader
 import org.locationtech.geomesa.utils.geotools._
-import org.opengis.feature.simple.SimpleFeatureType
+import org.opengis.feature.simple.{SimpleFeature, SimpleFeatureType}
 
+import scala.util.{Failure, Success}
 import scala.util.control.NonFatal
 
+object AvroIngestProcessor {
+  val AvroMatchMode: PropertyDescriptor =
+    new PropertyDescriptor.Builder()
+      .name("Avro SFT match mode")
+      .description("Determines how Avro SFT mismatches are handled")
+      .required(false)
+      .defaultValue("strict")
+      .allowableValues("strict", "lenient")
```

Comment:
instead of 'strict' and 'lenient', would it be clearer to use something like 'by attribute index' (or number/order?) and 'by attribute name'?

Reply:
If we uses strings with spaces, will the drop down be filled in with the options?

Reply:
i think that will still work ok.